app/src/main/java/com/imashnake/animite/features/searchbar/SearchViewModel.kt

```diff
@@ -21,7 +21,6 @@ class SearchViewModel @Inject constructor(
     var uiState by mutableStateOf(SearchUiState())
         private set
 
-    // TODO: Understand coroutines better.
     private var fetchJob: Job? = null
 
     fun searchAnime(search: String) {
```

Comment:
Me and Zhuin have mentioned it a few times, but for things like this we prefer a mapped flow from the savedStateHandle so it's stateful and survives process death like
```
@HiltViewModel
class SearchViewModel @Inject constructor(
    private val searchRepository: SearchRepository,
    private val savedStateHandle: SavedStateHandle
): ViewModel() {
  
  private val searchTerm = savedStateHandle.getStateFlow<String?>("search", null)

  val result = searchTerm
    .filterNotNull()
    .flatMapLatest { search -> 
        searchRepository.search(
          mediaType = MediaType.ANIME,
          perPage = 10,
            search = search
          )
    }
    .map(::SearchUiState)

    fun search(searchTerm: String) {
        savedStateHandle["search"] = searchTerm
    }
}
```
And then collectAsState etc

Reply:
Noted.